app/assets/javascripts/code/services/_fixed_sessions.js

```diff
@@ -116,8 +117,54 @@ export const fixedSessions = (
     },
 
     drawSessionsInLocation: function() {
-      map.markers = [];
-      (this.get()).forEach(session => drawSession.drawFixedSession(session));
+      if (sensors.anySelected() && this.scope.params.get('data').location.streaming) {
+        const sensorId = sensors.selectedId() || sensors.tmpSelectedId();
+        const sensor = sensors.sensors[sensorId] || {};
+        const selectedSensor = sensor.sensor_name;
+        (this.get()).forEach(session => this.drawColorCodedMarkers(session, selectedSensor));
+      } else {
+        (this.get()).forEach(session => this.drawDefaultMarkers(session));
+      }
+    },
+
+    drawColorCodedMarkers: function(session, selectedSensor) {
+      drawSession.undoDraw(session);
+      session.markers = [];
+
+      const content = Session.lastHourAverageVauleAndUnit(session, selectedSensor);
+      const heatLevel = heat.levelName(Session.lastHourAverage(session));
+      const latLng = Session.latLng(session);
+      const callback = (id) => () => $rootScope.$broadcast('markerSelected', {session_id: id});
+
+      if (!heat.outsideOfScope(heatLevel)) {
+        const customMarker = map.drawCustomMarker({
+            latLng: latLng,
+            content: content,
+            colorClass: heatLevel,
+            callback: callback(Session.id(session))
+          });
+        session.markers.push(customMarker);
+        map.markers.push(customMarker);
+        session.drawed = true;
```

Comment:
We could move all of the `session.` to `Session.`. What's left now prolly relies on the mutation so we cannot really make the update like in fp.

For example we could have
```
// Session.js
const addMarker = (session, marker) => ({ ...session, markers: [...session.markers, marker] })
```
that is we return a new session with the added marker and no mutations.

Unfortunately, we cannot do that most of the times. In fact, if there's a watch keeping an eye on the session, with no mutations it wouldn't trigger. That's a limit we have in Angular